src/firebolt/service/V1/region.py

```diff
@@ -0,0 +1,65 @@
+from typing import Dict, List
+
+from firebolt.model.V1.region import Region, RegionKey
+from firebolt.service.manager import ResourceManager
+from firebolt.service.V1.base import BaseService
+from firebolt.utils.urls import REGIONS_URL
+from firebolt.utils.util import cached_property
+
+
+class RegionService(BaseService):
+    def __init__(self, resource_manager: ResourceManager):
+        """
+        Service to manage AWS regions (us-east-1, etc)
+
+        Args:
+            resource_manager: Resource manager to use
+        """
+
+        super().__init__(resource_manager=resource_manager)
+
+    @cached_property
+    def regions(self) -> List[Region]:
+        """List of available AWS regions on Firebolt."""
+
+        response = self.client.get(url=REGIONS_URL, params={"page.first": 5000})
+        return [Region.parse_obj(i["node"]) for i in response.json()["edges"]]
+
+    @cached_property
+    def regions_by_name(self) -> Dict[str, Region]:
+        """Dict of {RegionLookup to Region}"""
+
+        return {r.name: r for r in self.regions}
+
+    @cached_property
+    def regions_by_key(self) -> Dict[RegionKey, Region]:
+        """Dict of {RegionKey to Region}"""
+
+        return {r.key: r for r in self.regions}
+
+    @cached_property
+    def default_region(self) -> Region:
+        """Default AWS region, could be provided from environment."""
+
+        if not self.default_region_setting:
+            raise ValueError(
+                "The environment variable FIREBOLT_DEFAULT_REGION must be set."
```

Comment:
Should we update this with `default_region variable should be provided`? I don't think we even support that env variable anymore